analysis/runtime/src/runtime/backend.rs

```diff
@@ -0,0 +1,95 @@
+use enum_dispatch::enum_dispatch;
+use fs_err::File;
+use std::env;
+use std::io::BufWriter;
+use std::sync::mpsc::Receiver;
+
+use bincode;
+
+use super::{AnyError, FINISHED};
+use crate::events::{Event, EventKind};
+use crate::metadata::Metadata;
+
+#[enum_dispatch]
+pub(super) trait IBackend {
```

Comment:
nit: can this just be `Backend`? All traits are interfaces.

Reply:
Yeah, I had tried to make it `Backend`, too, but `trait`s and `struct`/`enum`s can't overlap.  I thought about making the `trait` `Backend` and the `enum` `BackendImpl`, but then the `{Log,Debug}Backend`s should be `Impl`s, too, which I thought made less sense than naming this `IBackend`.

Reply:
Fixed in https://github.com/immunant/c2rust/pull/490/commits/e3098a14131f0acd39db71a5dbc3d0709fcb694f.